pkg/network/master/master.go

```diff
@@ -146,6 +146,27 @@ func Start(networkConfig osconfigapi.MasterNetworkConfig, networkClient networkc
 		return err
 	}
 
+	var nodes []kapi.Node
+	var subnets []networkapi.HostSubnet
+	if nodeList, err := master.kClient.Core().Nodes().List(metav1.ListOptions{}); err == nil {
+		nodes = nodeList.Items
+	}
+	if subnetList, err := master.networkClient.Network().HostSubnets().List(metav1.ListOptions{}); err == nil {
+		subnets = subnetList.Items
+	}
+
+	for _, subnet := range subnets {
+		for i, node := range nodes {
+			if subnet.Name == node.Name {
+				break
+			}
+			if i == len(nodes)-1 {
+				glog.Infof("Setup found no node associated with hostsubnet %s, deleting the hostsubnet", subnet.Name)
+				master.networkClient.Network().HostSubnets().Delete(subnet.Name, &metav1.DeleteOptions{})
+			}
+		}
+	}
```

Comment:
It will be good if you can encapsulate this code into a method deleteStaleHostSubnets()? and call this method from inside SubnetStartMaster() before we invoke `master.watchNodes()`.
SubnetStartMaster() already has subnet list, you just need to fetch the node list.